lib/plugins/body_reader.js

```diff
@@ -20,22 +20,27 @@ var MD5_MSG = 'Content-MD5 \'%s\' didn\'t match \'%s\'';
 ///--- Helpers
 
 function createBodyWriter(req) {
+    var buffers = [];
+
     var contentType = req.contentType();
+    var isText = false;
     if (!contentType ||
         contentType === 'application/json' ||
         contentType === 'application/x-www-form-urlencoded' ||
         contentType === 'multipart/form-data' ||
         contentType.substr(0, 5) === 'text/') {
-
-        req.body = '';
-        return (function (chunk) {
-            req.body += chunk.toString('utf8');
-        });
+        isText = true;
     }
 
     req.body = new Buffer(0);
     return (function (chunk) {
-        req.body = Buffer.concat([req.body, chunk]);
+        if (chunk)
```

Comment:
Please add braces around this if block -- since you've got braces around the else block